plugins/node/opentelemetry-hapi-instrumentation/src/utils.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Attributes } from '@opentelemetry/api';
+import { HttpAttribute } from '@opentelemetry/semantic-conventions';
+import type * as Hapi from '@hapi/hapi';
+import {
+  AttributeNames,
+  HapiLayerType,
+  HapiLifecycleMethodNames,
+} from './types';
+
+export function getPluginName<T>(plugin: Hapi.Plugin<T>): string {
+  if ((plugin as Hapi.PluginNameVersion).name) {
+    return (plugin as Hapi.PluginNameVersion).name;
+  } else {
+    return (plugin as Hapi.PluginPackage).pkg.name;
+  }
+}
+
+export const isLifecycleExtType = (
+  variableToCheck: unknown
+): variableToCheck is Hapi.ServerRequestExtType => {
+  return (
+    typeof variableToCheck === 'string' &&
+    HapiLifecycleMethodNames.includes(variableToCheck)
```

Comment:
Is this the only place `HapiLifecycleMethodNames` is used? If so, I think it should be an object instead to not do linear array searching

Reply:
Yes, this is. Did you mean something more like defining the type as:
```
export const HapiLifecycleMethodNames = {
  onPreAuth: 'onPreAuth',
  onCredentials: 'onCredentials',
  onPostAuth: 'onPostAuth',
  onPreHandler: 'onPreHandler',
  onPostHandler: 'onPostHandler',
  onPreResponse: 'onPreResponse',
  onRequest: 'onRequest'
 };
```
and then here when checking a given variable, instead checking `variableToCheck in HapiLifecycleMethodName`?

Otherwise, I am not sure if there is a cleaner way to perform this check (there was a [similar post on stackoverflow](https://stackoverflow.com/questions/36836011/checking-validity-of-string-literal-union-type-at-runtime) which I was looking at for reference. Please let me know if there's something I'm missing!

Reply:
Yes, that is what he meant. You could also have them just be `true`. The reason is to prevent having to search through an array each time, which is slower (`O(n)`) than an object access (`O(1)`).

Another option is to use a `Set`, which also has `O(1)` existence checking.

```typescript
const hapiLifecycleMethodNames = new Set([
  'onPreAuth',
  'onCredentials',
  'onPostAuth',
  // ... and so on
]);

console.log(hapiLifecycleMethodNames.has('onPreAuth')) // true
console.log(hapiLifecycleMethodNames.has('someOtherMethod')) // false
```

Reply:
Okay thanks, that makes sense!